app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cached_data = {}
+
+    def argument_based_cacher(*args) -> Any:
```

Comment:
You could use decorator from `functools` to preserve the wrapped function naming and docstring.